lib/constants.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Application constants.
+ * @type {{DIRECTORY_APP: string, DIRECTORY_APP_LAYOUTS: string, DIRECTORY_APP_PAGES: string,
```

Comment:
how about putting the type param with each line?
eg :
```javascript
var constants = {};
 /** @const @type {string} */ 
constants.DIRECTORY_AP =  "/app";
```

will idea find it ? if so that way is more maintainable.